Managers/UInputMan.cpp

```diff
@@ -839,7 +789,10 @@ namespace RTE {
 		}
 
 		if (g_InActivity) {
-			if (AnyStartPress(false) && !dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity())->IsBuyGUIVisible(-1)) {
+			const GameActivity *gameActivity = dynamic_cast<GameActivity *>(g_ActivityMan.GetActivity());
+			bool anyVisibleBuyGUI = (gameActivity) ? gameActivity->IsBuyGUIVisible(-1) : false;
+
+			if (AnyStartPress(false) && !anyVisibleBuyGUI) {
```

Comment:
You can simplify all this to
```cpp
const GameActivity *gameActivity = dynamic_cast<GameActivity *>(g_ActivityMan.GetActivity());
if (AnyStartPress(false) && (!gameActivity || !gameActivity->IsBuyGUIVisible(-1)) {
```
If you want very minor gainz when checking this while not ingame you could swap the order around to `if ((!gameActivity || !gameActivity->IsBuyGUIVisible(-1) && AnyStartPress(false)) {` but that makes it a little more nonsensical and the gainz aren't relevant cause they're tiny and only actually happen when not ingame.